src/figuresClasses.ts

```diff
@@ -1,19 +1,93 @@
-export interface Figure {
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
 
+enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue',
 }
 
-export class Triangle {
+export interface Figure {
+  shape: Shape;
+  color: Color;
 
+  getArea(): number;
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  shape: Shape.Triangle;
+
+  constructor(
+    public color: Figure['color'],
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    this.shape = Shape.Triangle;
+
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0) {
+      throw new Error('Please enter valid positive numbers');
+    }
+
+    if (this.a + this.b <= this.c
+      || this.a + this.c <= this.b
+      || this.c + this.b <= this.a) {
```

Comment:
Long expressions move to variables with meaningful variable naming. Fix all cases